application/helpers/expressions/em_manager_helper.php

```diff
@@ -7744,7 +7744,8 @@ public static function GetRelevanceAndTailoringJavaScript($bReturnArray=false)
                             $relParts[] = "  $('#question" . $arg['qid'] . "').trigger('relevance:on');\n";
                             if ($arg['type'] == 'S')
                             {
-                                $relParts[] = "  if($('#question" . $arg['qid'] . " div[id^=\"gmap_canvas\"]').length > 0)\n";
```

Comment:
For master : can we move this part on $(document).on("relevance:on") function ?

Reply:
Maybe in 3.X too in fact …

Reply:
Tested the issue by pulling the PR. Working as expected.

Reply:
Yes, it's not working or not working :) . It's Map question javascript must be included in Map question system. Not added in core (if we can)

Reply:
I can do an extra function inside maps.js that will do the setup taking a `qid` as parameter.
So that block will be moved into a new function.
Would that work?

Reply:
Remind : it's only my opinion :) 

Think adding here : https://github.com/LimeSurvey/LimeSurvey/blob/9fd5e78aa6a67b1078a05746c29bceb9f8fab7c0/assets/scripts/map.js#L18

```
$("#question"+question_id).on("relevance:on", function() { 
 resetMap(question_id);
});
```

But make it only in 4.X , and we can remove the EM part

Reply:
No need extra function : map.js have already function no need more

Reply:
Sorry. This has been tested and merged. Not sure about the idea. I don't dare to do that now.